brave/src/main/java/brave/internal/Lists.java

```diff
@@ -14,30 +14,32 @@
 package brave.internal;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 
 public final class Lists {
 
-  public static List<Object> ensureMutable(List<Object> list) {
+  public static <E> List<E> ensureMutable(List<E> list) {
     if (list instanceof ArrayList) return list;
     int size = list.size();
-    ArrayList<Object> mutable = new ArrayList<>(size);
+    ArrayList<E> mutable = new ArrayList<>(size);
     for (int i = 0; i < size; i++) {
       mutable.add(list.get(i));
     }
     return mutable;
   }
 
-  public static List<Object> ensureImmutable(List<Object> list) {
+  public static <E> List<E> ensureImmutable(List<E> list) {
     if (list.isEmpty()) return Collections.emptyList();
     // Faster to make a copy than check the type to see if it is already a singleton list
     if (list.size() == 1) return Collections.singletonList(list.get(0));
```

Comment:
I probably made this comment, it feels wrong :) anyway we can retouch this later.

Reply:
basically this implies reading the class name is worse than allocating a new wrapper.